### PR TITLE
Implement Rotations

### PR DESCRIPTION
**Pull request by OwenArnold**
* Implement rotations using Eigen
* Fix and extend readme and travis files
* Add `RotateCommand`
* Extend tests for rotations
* Extend write benchmark coverage

### CODE REVIEW

File: cow_instrument/testing/DetectorComponentTest.cpp [hunk 1]
```diff
@@ -65,9 +65,117 @@ TEST(detector_component_test, test_copy) {
...
+  const Eigen::Vector3d rotationCenter{0, 0, 0};
+
+  DetectorComponent detector(
+      ComponentIdType(1), DetectorIdType{1},
+      rotationCenter /*I make this the position of the detector*/);
+
+  // Rotate once by 90 degrees
+  detector.rotate(rotationAxis, rotationAngle, rotationCenter);
+  // Rotate again by 90 degrees
+  detector.rotate(rotationAxis, rotationAngle, rotationCenter);
```
Comment (SimonHeybrock) [inline cow_instrument/testing/DetectorComponentTest.cpp]: We should have tests here that test two additional things:

- The **ordering** of rotations is correct
- The **sign** of rotations is correct

With the parameters chosen here the test would not fail of any or both were wrong.

File: cow_instrument/CompositeComponent.cpp [hunk 2]
```diff
@@ -30,6 +26,27 @@ void CompositeComponent::shiftPositionBy(const V3D &delta) {
...
+}
+
+void CompositeComponent::rotate(const Eigen::Affine3d &transform,
+                                const Eigen::Quaterniond &rotationPart) {
+  for (size_t i = 0; i < m_children.size(); ++i) {
+
+    m_children[i]->rotate(transform, rotationPart);
+  }
+}
+
```
Comment (SimonHeybrock) [inline cow_instrument/CompositeComponent.cpp]: As just discussed on the whiteboard, we can test the other extreme of having a different parameters for every child component:

```cpp
void CompositeComponent::shiftChildPositionsBy(const std::vector<Vector3d> &deltas) {
  for (size_t i = 0; i < m_children.size(); ++i) {
     m_children[i]->shiftPositionBy(deltas[i]);
   }
}
```

Similarly for rotations.

A new command must be added that contains parameters for **all* pixels. Having a separate command for each pixel would add enormous overhead, since it will trigger a rebuild of, e.g., `DetectorInfo::m_L2s` once for each pixel.

Comment (OwenArnold) [inline cow_instrument/CompositeComponent.cpp]: I want to do this as Another PR as it will involve new commands

**Comment by OwenArnold** [comment]
Fixes #4
